### PR TITLE
sstables: Temporarily disable loading of first and last position metadata

### LINKED ISSUE #11642: oversized allocations due to find_first_position_in_partition
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/94/artifact/logs-full.release.004/1663300412868_test_cdc_large_values.py%3A%3ATestLargeColumnsWithCDC%3A%3Atest_large_blob_in_map_delta_preimage_full%5Bunprepared_statements%5D/node1.log
since scylla version 78850884d22f (that includes 9bcad9ffa89e61458dd18580bb9bd464c9770122)

For example:
```
WARN  2022-09-16 03:51:08,755 [shard 0] seastar_memory - oversized allocation: 1052672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at 0x517c21e 0x517c710 0x517ca18 0x4db33fa 0x4db4b2f 0x4db7755 0x1e39d00 0x1e3ab9e 0x4df2074 0x4df3457 0x4df26ac 0x4d98528 0x4d97a01 0x10ad07a 0x10aab85 /jenkins/workspace/scylla-master/dtest-daily-release/scylla/.ccm/scylla-repository/78850884d22fc4f84a13c683f2eaacc6a0e6b3bf/scylla/libreloc/libc.so.6+0x27b74 0x10a8cad
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >, seastar::input_stream<char>::read_exactly(unsigned long)::{lambda(auto:1)#1}, seastar::future<seastar::temporary_buffer<char> >::then_impl_nrvo<seastar::input_stream<char>::read_exactly(unsigned long)::{lambda(auto:1)#1}, seastar::future<seastar::temporary_buffer<char> > >(seastar::input_stream<char>::read_exactly(unsigned long)::{lambda(auto:1)#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >&&, seastar::input_stream<char>::read_exactly(unsigned long)::{lambda(auto:1)#1}&, seastar::future_state<seastar::temporary_buffer<char> >&&)#1}, seastar::temporary_buffer<char> >
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >, seastar::future<seastar::temporary_buffer<char> >::finally_body<sstables::sstable::data_read(unsigned long, unsigned long, seastar::io_priority_class const&, reader_permit)::$_49::operator()<seastar::input_stream<char> >(seastar::input_stream<char>&) const::{lambda()#1}, true>, seastar::future<seastar::temporary_buffer<char> >::then_wrapped_nrvo<seastar::future<seastar::temporary_buffer<char> >, seastar::future<seastar::temporary_buffer<char> >::finally_body<sstables::sstable::data_read(unsigned long, unsigned long, seastar::io_priority_class const&, reader_permit)::$_49::operator()<seastar::input_stream<char> >(seastar::input_stream<char>&) const::{lambda()#1}, true> >(seastar::future<seastar::temporary_buffer<char> >::finally_body<sstables::sstable::data_read(unsigned long, unsigned long, seastar::io_priority_class const&, reader_permit)::$_49::operator()<seastar::input_stream<char> >(seastar::input_stream<char>&) const::{lambda()#1}, true>&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >&&, seastar::future<seastar::temporary_buffer<char> >::finally_body<sstables::sstable::data_read(unsigned long, unsigned long, seastar::io_priority_class const&, reader_permit)::$_49::operator()<seastar::input_stream<char> >(auto:1&) const::{lambda()#1}, true>&, seastar::future_state<seastar::temporary_buffer<char> >&&)#1}, seastar::temporary_buffer<char> >
   --------
   seastar::internal::do_with_state<std::tuple<seastar::input_stream<char> >, seastar::future<seastar::temporary_buffer<char> > >
   --------
   seastar::internal::coroutine_traits_base<seastar::input_stream<char> >::promise_type
   --------
   seastar::internal::coroutine_traits_base<seastar::temporary_buffer<char> >::promise_type
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<seastar::bool_class<seastar::stop_iteration_tag> >, seastar::input_stream<char>::consume<std::reference_wrapper<sstables::mx::data_consume_rows_context_m> >(std::reference_wrapper<sstables::mx::data_consume_rows_context_m>&&)::{lambda()#1}::operator()()::{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::temporary_buffer<char> >::then_impl_nrvo<seastar::input_stream<char>::consume<std::reference_wrapper<sstables::mx::data_consume_rows_context_m> >(std::reference_wrapper<sstables::mx::data_consume_rows_context_m>&&)::{lambda()#1}::operator()()::{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::bool_class<seastar::stop_iteration_tag> > >(std::reference_wrapper<sstables::mx::data_consume_rows_context_m>&&)::{lambda(seastar::internal::promise_base_with_type<seastar::bool_class<seastar::stop_iteration_tag> >&&, seastar::input_stream<char>::consume<std::reference_wrapper<sstables::mx::data_consume_rows_context_m> >(auto:1&&)::{lambda()#1}::operator()()::{lambda(seastar::temporary_buffer<char>)#1}&, seastar::future_state<seastar::temporary_buffer<char> >&&)#1}, seastar::temporary_buffer<char> >
   --------
   seastar::internal::repeater<seastar::input_stream<char>::consume<std::reference_wrapper<sstables::mx::data_consume_rows_context_m> >(std::reference_wrapper<sstables::mx::data_consume_rows_context_m>&&)::{lambda()#1}>
   --------
   seastar::internal::do_until_state<sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda()#3}::operator()() const::{lambda()#1}, sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda()#3}::operator()() const::{lambda()#2}>
   --------
   seastar::internal::do_until_state<sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda()#2}, sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda()#3}>
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda(seastar::future<void>)#1}, seastar::future<void>::then_wrapped_nrvo<void, sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda(seastar::future<void>)#1}>(sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda(seastar::future<void>)#1}&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, sstables::mx::mx_sstable_mutation_reader::fill_buffer()::{lambda(seastar::future<void>)#1}&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<seastar::bool_class<seastar::stop_iteration_tag> >, flat_mutation_reader_v2::impl::consume_pausable<std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> > >(std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> >)::{lambda()#1}::operator()()::{lambda()#1}, seastar::future<void>::then_impl_nrvo<flat_mutation_reader_v2::impl::consume_pausable<std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> > >(std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> >)::{lambda()#1}::operator()()::{lambda()#1}, seastar::future<seastar::bool_class<seastar::stop_iteration_tag> > >(flat_mutation_reader_v2::impl::consume_pausable<std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> > >(std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> >)::{lambda()#1}::operator()()::{lambda()#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::bool_class<seastar::stop_iteration_tag> >&&, flat_mutation_reader_v2::impl::consume_pausable<std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> > >(auto:1)::{lambda()#1}::operator()()::{lambda()#1}&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
   --------
   seastar::internal::repeater<flat_mutation_reader_v2::impl::consume_pausable<std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> > >(std::reference_wrapper<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> >)::{lambda()#1}>
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<seastar::optimized_optional<mutation> >, flat_mutation_reader_v2::impl::consume<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>(sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder)::{lambda(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&)#1}::operator()(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&) const::{lambda()#1}, seastar::future<void>::then_impl_nrvo<flat_mutation_reader_v2::impl::consume<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>(sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder)::{lambda(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&)#1}::operator()(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&) const::{lambda()#1}, seastar::future<seastar::optimized_optional<mutation> > >(flat_mutation_reader_v2::impl::consume<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>(sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder)::{lambda(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&)#1}::operator()(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&) const::{lambda()#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::optimized_optional<mutation> >&&, flat_mutation_reader_v2::impl::consume<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>(auto:1)::{lambda(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&)#1}::operator()(flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder>&) const::{lambda()#1}&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
   --------
   seastar::internal::do_with_state<std::tuple<flat_mutation_reader_v2::impl::consumer_adapter<sstables::sstable::find_first_position_in_partition(reader_permit, dht::decorated_key const&, bool, seastar::io_priority_class const&)::position_finder> >, seastar::future<seastar::optimized_optional<mutation> > >
   --------
   seastar::internal::coroutine_traits_base<std::optional<position_in_partition> >::promise_type
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
```

### PR DESCRIPTION
**Pull request by raphaelsc**
It's known that reading large cells in reverse cause large allocations. Source: https://github.com/scylladb/scylladb/issues/11642

The loading is preliminary work for splitting large partitions into fragments composing a run and then be able to later read such a run in an efficiency way using the position metadata.

The splitting is not turned on yet, anywhere. Therefore, we can temporarily disable the loading, as a way to avoid regressions in stable versions. Large allocations can cause stalls due to foreground memory eviction kicking in.
The default values for position metadata say that first and last position include all clustering rows, but they aren't used anywhere other than by sstable_run to determine if a run is disjoint at clustering level, but given that no splitting is done yet, it does not really matter.

Unit tests relying on position metadata were adjusted to enable the loading, such that they can still pass.

Fixes #11642.

### CODE REVIEW

**Comment by raphaelsc** [comment]
This is targeting branch-5.2. Once I am back next Friday, will be able to make progress on the actual fix.

File: test/boost/sstable_compaction_test.cc [hunk 1]
```diff
@@ -4946,6 +4946,7 @@ SEASTAR_TEST_CASE(test_large_partition_splitting_on_compaction) {
         position_in_partition::tri_compare pos_tri_cmp(*s);
 
         for (auto& sst : ret.new_sstables) {
+            sst = env.reusable_sst(s, tmp.path().string(), sst->generation().value()).get0();
```
Comment (denesb) [inline test/boost/sstable_compaction_test.cc:4949]: Why is this necessary?

**Review by denesb — APPROVED** [review]
LGTM, one question.

File: test/boost/sstable_compaction_test.cc (hunk 1 above)
Comment (bhalevy) [reply test/boost/sstable_compaction_test.cc:4949]: Quoting from the commit message:
```
Unit tests relying on position metadata were adjusted to enable the loading, such that they can still pass.
```

`reusable_sst` below is changes to pass `{ .load_first_and_last_position_metadata = true }` to `sst->load()`

**Review by bhalevy — APPROVED** [review]

File: test/boost/sstable_compaction_test.cc (hunk 1 above)
Comment (denesb) [reply test/boost/sstable_compaction_test.cc:4949]: Ah, I missed that.

**Comment by denesb** [comment]
> This is targeting branch-5.2. Once I am back next Friday, will be able to make progress on the actual fix.

You mean it should be backported to branch-5.2? The target of the PR is master.

**Comment by raphaelsc** [comment]
> > This is targeting branch-5.2. Once I am back next Friday, will be able to make progress on the actual fix.
> 
> You mean it should be backported to branch-5.2? The target of the PR is master.

Yes, we can merge it to both master and then later branch-5.2. I can revert this patch on master once I have the proper fix for reading large cells in reverse.

**Comment by denesb** [comment]
Backported to 5.2.
